magicblock-chainlink/src/submux/subscription_task.rs

```diff
@@ -48,31 +89,21 @@ impl AccountSubscriptionTask {
 
             let mut errors = Vec::new();
             let mut tx = Some(tx);
-            let op_name = match self {
-                Subscribe(_) => "Subscribe",
-                Unsubscribe(_) => "Unsubscribe",
-                Shutdown => "Shutdown",
-            };
+            let mut successes = 0;
+            let op_name = self.op_name();
 
             while let Some((i, result)) = futures.next().await {
                 match result {
                     Ok(_) => {
-                        if let Some(tx) = tx.take() {
-                            let _ = tx.send(Ok(()));
+                        successes += 1;
+                        if successes >= target_successes {
+                            if let Some(tx) = tx.take() {
+                                let _ = tx.send(Ok(()));
```

Comment:
nit: this sends the Ok(()) multiple times potentially, on every success after target has been reached